test/model/content_test.dart

```diff
@@ -1,16 +1,261 @@
+  /// conveniently getting such HTML.
+  final String html;
+
+  /// The [ZulipContent.nodes] expected from parsing [html].
+  final List<BlockContentNode> expectedNodes;
+
+  /// The text, if applicable, of a text widget expected from
+  /// rendering [expectedNodes].
+  ///
+  /// Strictly this belongs to the widget tests, not the model tests, as it
+  /// encodes choices about how the content widgets work.  But it's convenient
+  /// to have it defined for each test case right next to [html] and [expectedNodes].
+  final String? expectedText;
+
+  static final emojiUnicode = ContentExample.inline(
+    'Unicode emoji, encoded in span element',
+    ":thumbs_up:",
+    expectedText: '\u{1f44d}', // "👍"
+    '<p><span aria-label="thumbs up" class="emoji emoji-1f44d" role="img" title="thumbs up">:thumbs_up:</span></p>',
+    const UnicodeEmojiNode(emojiUnicode: '\u{1f44d}'));
+
+  static final emojiUnicodeClassesFlipped = ContentExample.inline(
+    'Unicode emoji, encoded in span element, class order reversed',
+    null, // ":thumbs_up:" (hypothetical server variation)
+    expectedText: '\u{1f44d}', // "👍"
+    '<p><span aria-label="thumbs up" class="emoji-1f44d emoji" role="img" title="thumbs up">:thumbs_up:</span></p>',
+    const UnicodeEmojiNode(emojiUnicode: '\u{1f44d}'));
+
+  static final emojiUnicodeMultiCodepoint = ContentExample.inline(
+    'Unicode emoji, encoded in span element, multiple codepoints',
+    ":transgender_flag:",
+    expectedText: '\u{1f3f3}\u{fe0f}\u{200d}\u{26a7}\u{fe0f}', // "🏳️‍⚧️"
+    '<p><span aria-label="transgender flag" class="emoji emoji-1f3f3-fe0f-200d-26a7-fe0f" role="img" title="transgender flag">:transgender_flag:</span></p>',
+    const UnicodeEmojiNode(emojiUnicode: '\u{1f3f3}\u{fe0f}\u{200d}\u{26a7}\u{fe0f}'));
+
+  static final emojiUnicodeLiteral = ContentExample.inline(
+    'Unicode emoji, not encoded in span element',
+    "\u{1fabf}",
+    expectedText: '\u{1fabf}', // "🪿"
+    '<p>\u{1fabf}</p>',
+    const TextNode('\u{1fabf}'));
+
+  static final emojiCustom = ContentExample.inline(
+    'custom emoji',
+    ":flutter:",
+    '<p><img alt=":flutter:" class="emoji" src="/user_avatars/2/emoji/images/204.png" title="flutter"></p>',
+    const ImageEmojiNode(
+      src: '/user_avatars/2/emoji/images/204.png', alt: ':flutter:'));
+
+  static final emojiZulipExtra = ContentExample.inline(
+    'Zulip extra emoji',
+    ":zulip:",
+    '<p><img alt=":zulip:" class="emoji" src="/static/generated/emoji/images/emoji/unicode/zulip.png" title="zulip"></p>',
+    const ImageEmojiNode(
+      src: '/static/generated/emoji/images/emoji/unicode/zulip.png', alt: ':zulip:'));
+
+  static const quotation = ContentExample(
+    'quotation',
+    "```quote\nwords\n```",
+    expectedText: 'words',
+    '<blockquote>\n<p>words</p>\n</blockquote>', [
+        QuotationNode([ParagraphNode(links: null, nodes: [TextNode('words')])])
+      ]);
```

Comment:
nit: this should get just one level of indentation
```suggestion
    '<blockquote>\n<p>words</p>\n</blockquote>', [
      QuotationNode([ParagraphNode(links: null, nodes: [TextNode('words')])])
    ]);
```